### PR TITLE
DAC6-2808 | AddressLookup and SelectAddressPage implementation

### PR DESCRIPTION
**Pull request by matthewharris1**

### CODE REVIEW

File: app/forms/InstitutionSelectAddressFormProvider.scala [hunk 1]
```diff
@@ -16,17 +16,16 @@
...
 import forms.mappings.Mappings
 import play.api.data.Form
-import models.InstitutionSelectAddress
+
+import javax.inject.Inject
 
 class InstitutionSelectAddressFormProvider @Inject() extends Mappings {
 
-  def apply(): Form[InstitutionSelectAddress] =
+  def apply(): Form[String] =
```
Comment (C-Thomp) [inline app/forms/InstitutionSelectAddressFormProvider.scala:26]: Should this form take the Address type from models if it has multiple lines of information, or will it always be a single line string?

File: app/controllers/InstitutionSelectAddressController.scala [hunk 2]
```diff
@@ -48,26 +52,57 @@ class InstitutionSelectAddressController @Inject() (
...
-        case Some(value) => form.fill(value)
-      }
+      request.userAnswers.get(AddressLookupPage) match {
+        case Some(addresses) =>
+          val preparedForm: Form[String] = request.userAnswers.get(InstitutionSelectAddressPage) match {
+            case None        => form
+            case Some(value) => form.fill(value)
+          }
+
+          val radios: Seq[RadioItem] = addresses.map(
```
Comment (C-Thomp) [inline app/controllers/InstitutionSelectAddressController.scala:62]: can we use a more descriptive name for 'radios'? These are the returned address from the lookup service right?

File: app/models/Regime.scala
Comment (C-Thomp) [inline app/models/Regime.scala:1]: Is this file needed? seems like the regime in the lookup post just needs to be a string of "CRSFATCA"

File: app/forms/InstitutionSelectAddressFormProvider.scala (hunk 1 above)
Comment (matthewharris1) [reply app/forms/InstitutionSelectAddressFormProvider.scala:26]: I believe this will always be a single line string but will be good to test this when 2707 is done to double check

File: app/controllers/InstitutionSelectAddressController.scala (hunk 2 above)
Comment (matthewharris1) [reply app/controllers/InstitutionSelectAddressController.scala]: Change name to radioAddressFields

File: app/models/Regime.scala
Comment (matthewharris1) [reply app/models/Regime.scala:1]: Yeah you're right i've removed this and replace with "CRSFATCA" in the connector
